src/Rules/Functions/PrintfParametersRule.php

```diff
@@ -40,25 +50,17 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		$functionsArgumentPositions = [
-			'printf' => 0,
-			'sprintf' => 0,
-			'sscanf' => 1,
-			'fscanf' => 1,
-		];
-		$minimumNumberOfArguments = [
-			'printf' => 1,
-			'sprintf' => 1,
-			'sscanf' => 3,
-			'fscanf' => 3,
-		];
-
-		$name = strtolower((string) $node->name);
-		if (!array_key_exists($name, $functionsArgumentPositions)) {
+		if (!$this->reflectionProvider->hasFunction($node->name, $scope)) {
```

Comment:
Why are you refactoring this so much? Either revert it, or send a PR to merge before this one :)